react/PlaceOrder.tsx

```diff
@@ -94,18 +109,48 @@ const PlaceOrder: React.FC = () => {
         []
       )
 
-      const { data: redirectUrl } = await postRobot.send(
-        cardFormIframe.contentWindow,
-        'sendPayments',
-        {
-          payments: allPayments,
-          receiverUri,
-          orderId: orderGroupId,
-          gatewayCallbackTemplatePath,
-          transactionId,
-        }
+      const hasSensitiveData = allPayments.some(payment =>
+        paymentsWithSensitiveData.includes(payment.id)
       )
 
+      let redirectUrl
+
+      if (hasSensitiveData) {
+        const { data } = await postRobot.send(
+          cardFormIframe.contentWindow,
+          'sendPayments',
+          {
+            payments: allPayments,
+            receiverUri,
+            orderId: orderGroupId,
+            gatewayCallbackTemplatePath,
+            transactionId,
+          }
+        )
+
+        redirectUrl = data.redirectUrl
```

Comment:
`data` is the redirect URL itself, not an object with a `redirectUrl` property. That explains why after completing a purchase with a new credit card I'm redirected to `/undefined` 😆 

```suggestion
        redirectUrl = data
```

I actually recommend renaming `data` to something else, maybe `url`.

Reply:
I noticed that 😄 I'll rename it :-)

Reply:
done in b9cd926